### ISSUE TITLE
Join forces?

### ISSUE THREAD

**Issue by brynbellomy** [colinta/Swift-for-f-ing-sublime#1]
Before anything, the tone of this repo is hilarious and reminds me of some similarly aggressive READMEs I've written :+1: :laughing: 

I've been working on a similar revamp of the Swift syntax that quiqueg put out a while back.  It's a complete rewrite at this point, and I sort of used the C# syntax as a jumping-off point (and took an occasional look at the Java syntax for inspiration as well).

I've been checking out your syntax and I was wondering if you might want to take a look at the one I've been working on and maybe merge our efforts.

More generally, I'm sort of aiming at making a suite of Sublime plugins for Swift so that I can eventually avoid Xcode altogether.  If you have any interest in getting something like that going, let me know.  I have a working (but unpublished) plugin that replaces the built-in Sublime Goto functionality (it matches a few regexes against the source file and then opens a navigation popup similar to the Goto popup).  I'm also starting to work on new approach that leverages [jpsim/SourceKitten](https://github.com/jpsim/SourceKitten) to tap into information from SourceKitService.  If SourceKitten pans out, it could be the basis of a full IDE in Sublime (in the style of the incredible [Phaiax/ArcticTypescript](https://github.com/Phaiax/ArcticTypescript) plugin).

Anyway, here's the `YAML-.tmLanguage` file: https://github.com/brynbellomy/SublimeSwiftIDE/blob/master/New%20Swift.YAML-tmLanguage

... and the `.tmLanguage` file: https://github.com/brynbellomy/SublimeSwiftIDE/blob/master/New%20Swift.tmLanguage

Thanks!

**Comment by colinta** [comment]
Haha glad you dig the name - I was frustrated with the other swift packages at the time.

I found another one recently that is an improvement, best yet I think. Search for "Swift tmLanguage".

I'd definitely like to work on something cooperatively, but I'd like to take a stab at using the new sublime-syntax syntax. I'm not very proficient at the tmLanguage format. What do you use to convert it from YAML?

I'm struggling to find much time for open source projects these days, but I'm going to check out SourceKitten, I've never heard of it! I'll try to chip in where I can. 😄

**Comment by colinta** [comment]
Hey dude, I got a bunch of work done on the conversion to `.sublime-syntax`!  <https://github.com/colinta/Swift-for-f-ing-sublime/tree/sublime-syntax>

**Comment by brynbellomy** [comment]
Nice!  I haven't dug into that new file format yet; it's basically just a YAML-ized version of the original tmLanguage format, right?  If so, we'd have a much easier time merging some of the declarations from our two syntax files.

For YAML conversion I've been using the "convert yaml/json/plist" functionality in AAAPackageDev.

I'll give this a spin today.

**Comment by colinta** [comment]
It’s actually a *new syntax*, which makes it (for bettor or worse) incompatible with the tried-and-not-so-great tmLanguage format.  I find it to be much easier; this syntax file is barely over 200 lines.

**Comment by andreyvit** [comment]
Hey, @colinta, just a note that I couldn't get AAAPackageDev conversion to work in the latest HEAD. Getting:

    Input type not specified, auto-detecting...
    Unable to detect file type.

despite all my attempts. (And I even got the YAML to parse using yaml2json, which required removal of you `#######...######` separators. But that didn't help AAAPackageDev detect it.)

**Comment by colinta** [comment]
Does AAAPackageDev already support the new 'sublime-syntax' file format?

**Comment by andreyvit** [comment]
@colinta I assumed that AAAPackageDev has invented `.sublime-syntax`. No? I've only installed it today for the specific purpose of trying to compile your syntax file. Ended up using @brynbellomy's precompiled file for now.

Btw, FYI, when your plugin is installed via Package Control, it does not come with a compiled file and thus doesn't work. I spent a lot of time troubleshooting before I figured out that `.sublime-syntax` is not a natively supported format. :-)

**Comment by colinta** [comment]
Are you using ST2?  This is a ST3 feature.  I can create an st2 branch to
support the old .tmLanguage format.

**Comment by andreyvit** [comment]
@colinta No, I've long switched to ST3. Are you saying that it _is_ a native feature?

**Comment by colinta** [comment]
Hmm, weird that it doesn't work, then.  Yes, native feature, one of the most recent builds included it. <https://www.sublimetext.com/docs/3/syntax.html>

**Comment by colinta** [comment]
@andreyvit if you still have trouble, please open a new issue.  We kind of hijacked the conversation about new Swift packages! :smiley:

**Comment by andreyvit** [comment]
@colinta Yes, I'm sorry about that. I made a lot of wrong assumptions and thought I was continuing an ongoing discussion about converting to sublime-syntax.

On the topic of this issue, I'm very interested in what you guys can do, and would love to help occasionally (I used to make IDEs for a living). I've looked at SourceKitten, but it seems to re-parse the files on every invocation, which is not a very good foundation for code completion. :-)

**Comment by aristidesfl** [comment]
Hi guys, is this package supposed to be better than the more popular https://packagecontrol.io/packages/Swift ?

In which ways?

Thanks!

**Comment by colinta** [comment]
Try it out for yourself and let me know!

**Comment by aristidesfl** [comment]
I'm just learning Swift, I wouldn't know. Just using it for a small project. Would be happy to compare it it was a language I use consistently.

**Comment by colinta** [comment]
Fair enough, well then my answer is "yes, of course it's better, why else would I make it?" but that's not very helpful.

Do you use the "go to symbol" feature?  In that case I think you'll prefer mine.  Also I highlight class names and a few more things than that package.  I think the only feature that package has over mine is that it has a better name.

**Comment by aristidesfl** [comment]
It was not clear which one came first because package control only shows the number of installs in the search results. It would be a good improvment. 
![screen shot 2017-03-03 at 00 33 45](https://cloud.githubusercontent.com/assets/208149/23531941/361d1b5a-ffa9-11e6-83fc-e3331b106156.png)

Which other packages would recommend from that list? SwiftKitten and the linter?

**Comment by colinta** [comment]
No idea, I don't use any of em.  Sorry I'm not much help - you should try them out!

**Comment by aristidesfl** [comment]
Thanks colinta, sorry for abusing this thread. Anything else to enhance to improve the experience with / without xcode?
@andreyvit what about you, any experience on this?

**Comment by fufu930** [comment]
7 years had passed, but nevertheless thanks for this very useful project 😊
